Yasgui/packages/yasqe/src/sparql.ts

```diff
@@ -50,6 +50,51 @@ export function getAjaxConfig(
 }
 
 export async function executeQuery(yasqe: Yasqe, config?: YasqeAjaxConfig): Promise<any> {
+  if (yasqe.isUpdateQuery()) {
+    return executeUpdateModeQuery(yasqe, config);
+  }
+  return executeQueryModeQuery(yasqe, config);
+}
+
+export async function executeUpdateModeQuery(yasqe: Yasqe, config?: YasqeAjaxConfig): Promise<any> {
+  let initialRepositoryStatementsCount: number;
+  const checkQueryPreconditions = (): Promise<BeforeUpdateQuery> => {
+    return yasqe.config.beforeUpdateQuery();
+  };
+
+  const getRepositoryStatementsCountBeforeQuery = (): Promise<number> => {
+    return yasqe.config.getRepositoryStatementsCount().then((initialCount: number) => {
+      initialRepositoryStatementsCount = initialCount;
+      return initialCount;
+    });
+  };
+
+  const executeQuery = (): Promise<any> => {
+    return executeQueryModeQuery(yasqe, config);
+  };
+
+  const calculateAffectedStatementsCount = () => {
+    yasqe.config.getRepositoryStatementsCount().then((repositoryStatementsCount) => {
+      if (initialRepositoryStatementsCount !== undefined && repositoryStatementsCount !== undefined) {
+        const affectedStatementsCount = repositoryStatementsCount - initialRepositoryStatementsCount;
+        yasqe.emit("countAffectedRepositoryStatementsChanged", affectedStatementsCount);
+        yasqe.emit("countAffectedRepositoryStatementsPersisted");
+      }
+    });
+  };
+
+  return await checkQueryPreconditions()
```

Comment:
It's quite confusing when promises and async-await approaches are intermixed. The implementation in yasgui primarily relies on async-await so you can stick with it or just use promises and don't mix them up.